src/ui/osx/TogglDesktop/Feature/TimeEntry/TimeEntryDatasource.swift

```diff
@@ -272,6 +283,11 @@ extension TimeEntryDatasource {
                                 forSupplementaryViewOfKind:NSCollectionView.elementKindSectionHeader,
                                 withIdentifier: Constants.TimeHeaderID)
     }
+
+    fileprivate func lastDateInSections() -> Date? {
+        let lastSection = sections.reversed().first { !$0.isLoadMore }
+        return lastSection?.entries.last?.ended
```

Comment:
the day of time entry is assigned by the start time AFAIK, perhaps using `started` instead of `ended` is better here?

Reply:
It's good point. I've just updated 👍